### PR TITLE
feat: Add cache step for docker build

### PR DESCRIPTION
**Pull request by bradbeam**
Signed-off-by: Brad Beam <brad.beam@carbonrelay.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,13 +77,23 @@ generate: controller-gen
...
+		--target cache \
+		--cache-from=${CACHE_IMG}
+	docker build . -t ${IMG} \
+		--build-arg LDFLAGS='${LDFLAGS}' \
+		--cache-from=${CACHE_IMG}
+	docker build config -t ${SETUPTOOLS_IMG} \
+		--build-arg IMG='${IMG}' \
+		--build-arg PULL_POLICY='${PULL_POLICY}' \
+		--build-arg VERSION='${VERSION}' \
+		--cache-from=${CACHE_IMG}
```
Comment (jgustie) [inline Makefile]: You don't need to use the cache image for the setuptools.

File: .github/workflows/push.yaml [hunk 2]
```diff
@@ -33,6 +33,7 @@ jobs:
           echo "::set-env name=IMG::gcr.io/${{ secrets.GOOGLE_PROJECT_ID }}/redskyops-controller:${TAG}"
           echo "::set-env name=REDSKYCTL_IMG::gcr.io/${{ secrets.GOOGLE_PROJECT_ID }}/redskyctl:${TAG}"
           echo "::set-env name=SETUPTOOLS_IMG::gcr.io/${{ secrets.GOOGLE_PROJECT_ID }}/setuptools:${TAG}"
+          echo "::set-env name=CACHE_IMG::gcr.io/${{ secrets.GOOGLE_PROJECT_ID }}/redskyops-cache:latest"
```
Comment (jgustie) [inline .github/workflows/push.yaml:50]: This would need to be duplicated in the main workflow.

**Comment by jgustie** [comment]
Can you also pull in the "Cache Go Modules" step from the master workflow? I think I forgot to add that to the pull workflow.

Push the change to remove the `--cache-from` on the setup tools first though: I would like to see the timings strictly from the Docker cache image first.

File: .github/workflows/master.yaml [hunk 3]
```diff
@@ -17,10 +17,6 @@ jobs:
         uses: actions/checkout@v2
       - name: Unshallow
         run: git fetch --prune --unshallow
-      - name: Set up Go
```
Comment (bradbeam) [inline .github/workflows/master.yaml]: https://github.com/actions/virtual-environments/blob/8491d71a0b76ae4a8572a27f683e396269bfa80c/images/linux/ubuntu1804.json#L27-L28 // https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1804-README.md

Looks like go 1.14 is included in the runner by default, so we get to skip a step here.

Comment (bradbeam) [reply .github/workflows/master.yaml]: Eh, added this back in. There's a bit of a lag ( 1.14.1 vs 1.14.2 )

**Comment by bradbeam** [comment]
closing for now since there were no real benefits

**Pull request closed without merging**